src/components/application_manager/test/hmi_capabilities_test.cc

```diff
@@ -493,8 +502,34 @@ TEST_F(HMICapabilitiesTest, LoadCapabilitiesFromFile) {
                                                .asInt()));
     }
   }
+}
+
+TEST_F(HMICapabilitiesTest,
+       LoadCapabilitiesFromFile_CheckTextFields_SuccessLoadAndConvert) {
+  SetUpLanguageAndLightCapabilitiesExpectation();
+  hmi_capabilities_->Init(last_state_wrapper_);
+  const auto display_capabilities_so =
+      *(hmi_capabilities_->display_capabilities());
+
+  const uint32_t text_len =
+      display_capabilities_so[hmi_response::text_fields].length();
+  EXPECT_NE(0u, text_len);
+  for (uint32_t i = 0; i < text_len; ++i) {
+    const int32_t index = static_cast<int32_t>(i);
```

Comment:
@ydementieiev Possible type overflow.

Reply:
@AByzhynar removed this [in this PR](https://github.com/LuxoftSDL/sdl_core/pull/17/commits/53f85366b6a7cfc3fc2ccd60e33886690db075de)